internal/app/appmain.go

```diff
@@ -30,8 +30,8 @@ var (
 )
 
 // RunApplication creates a server.
-func RunApplication(serverName string, bindService func(*rpc.ServerParams, config.View) error) {
-	cfg, err := config.Read()
```

Comment:
I see the issue here  (or even in the existing implementation) of the RunApplication depending on values that may just not be set. 

One approach could be for it to define explicitly, a struct with the values it expects - and the callers to fill in that before calling RunApplication. Atleast from readability perspective we could then have a clear understanding of the values the caller chose to not to populate and document its impact.

Wont block this PR for it but can you create an issue for refactoring this?

Reply:
Done.